ui/src/app/workflows/components/resubmit-workflow-panel.tsx

```diff
@@ -12,103 +12,93 @@ interface Props {
-            memoized: false,
-            isSubmitting: false,
-            overrideParameters: false
+    const submit = () => {
+        setIsSubmitting(true);
+        const parameters: ResubmitOpts['parameters'] = overrideParameters
+            ? [...workflowParameters.filter(p => Utils.getValueFromParameter(p) !== undefined).map(p => p.name + '=' + Utils.getValueFromParameter(p))]
+            : [];
+        const opts: ResubmitOpts = {
+            parameters,
+            memoized
         };
-        this.state = state;
-    }
 
-    public render() {
-        return (
-            <>
-                <h4>Resubmit Workflow</h4>
-                <h5>
-                    {this.props.workflow.metadata.namespace}/{this.props.workflow.metadata.name}
-                </h5>
-                {this.state.error && <ErrorNotice error={this.state.error} />}
-                <div className='white-box'>
-                    <div key='override-parameters' style={{marginBottom: 25}}>
-                        <label>Override Parameters</label>
-                        <div className='columns small-9'>
-                            <Checkbox checked={this.state.overrideParameters} onChange={overrideParameters => this.setState({overrideParameters})} />
-                        </div>
-                    </div>
+        const handleSuccess = (submitted: Workflow) => {
+            document.location.href = uiUrl(`workflows/${submitted.metadata.namespace}/${submitted.metadata.name}`);
+        };
 
-                    {this.state.overrideParameters && (
-                        <div key='parameters' style={{marginBottom: 25}}>
-                            <label>Parameters</label>
-                            {this.state.workflowParameters.length > 0 && (
-                                <ParametersInput parameters={this.state.workflowParameters} onChange={workflowParameters => this.setState({workflowParameters})} />
-                            )}
-                            {this.state.workflowParameters.length === 0 ? (
-                                <>
-                                    <br />
-                                    <label>No parameters</label>
-                                </>
-                            ) : (
-                                <></>
-                            )}
-                        </div>
-                    )}
+        const handleError = (err: Error) => {
+            setError(err);
+            setIsSubmitting(false);
+        };
+
+        if (!props.isArchived) {
+            services.workflows
+                .resubmit(props.workflow.metadata.name, props.workflow.metadata.namespace, opts)
+                .then(handleSuccess)
+                .catch(handleError);
+        } else {
+            services.workflows
+                .resubmitArchived(props.workflow.metadata.uid, props.workflow.metadata.namespace, opts)
+                .then(handleSuccess)
+                .catch(handleError);
```

Comment:
```suggestion
        let submitted;
        try {
            if (!props.isArchived) {
                submitted = await services.workflows.resubmit(props.workflow.metadata.name, props.workflow.metadata.namespace, opts);
            } else {
                submitted = await services.workflows.resubmitArchived(props.workflow.metadata.uid, props.workflow.metadata.namespace, opts);
           }
           handleSuccess(submitted);
       } catch (newError) {
         handleError(newError);
       }
```

would be good to rewrite this with async/await while we're at it, which also makes it cleaner.

we can actually remove the `handleSuccess` and `handleError` functions and instead in-line them this way. `submit` just has to be an `async` function.

Reply:
could also set a variable for the function name as `retry` does instead of the `submitted` variable. that variant is maybe cleaner actually?

Reply:
```suggestion
        try {
            const submitted = !props.isArchived
                ? await services.workflows.resubmit(props.workflow.metadata.name, props.workflow.metadata.namespace, opts);
                : submitted = await services.workflows.resubmitArchived(props.workflow.metadata.uid, props.workflow.metadata.namespace, opts);
           handleSuccess(submitted);
       } catch (newError) {
           handleError(newError);
       }
```

yea the ternary is definitely cleaner